packages/context/src/interceptor.ts

```diff
@@ -48,14 +48,38 @@ export class InterceptedInvocationContext extends InvocationContext {
    */
   getGlobalInterceptorBindingKeys(): string[] {
     const bindings: Readonly<Binding<Interceptor>>[] = this.find(
-      filterByTag(ContextTags.GLOBAL_INTERCEPTOR),
+      binding =>
+        filterByTag(ContextTags.GLOBAL_INTERCEPTOR)(binding) &&
+        // Only include interceptors that match the source type of the invocation
+        this.applicableTo(binding),
     );
     this.sortGlobalInterceptorBindings(bindings);
     const keys = bindings.map(b => b.key);
     debug('Global interceptor binding keys:', keys);
     return keys;
   }
 
+  /**
+   * Check if the binding for a global interceptor matches the source type
+   * of the invocation
+   * @param binding - Binding
+   */
+  private applicableTo(binding: Readonly<Binding<unknown>>) {
+    const source = this.source && this.source.type;
```

Comment:
Ditto, should we use `??`

Reply:
```
src/interceptor.ts:68:35 - error TS2532: Object is possibly 'undefined'.

68     const source = this.source ?? this.source.type;
                                     ~~~~~~~~~~~
```

Reply:
Sorry, I suggested a wrong operator. I am still learning the ropes here...

I think the following is the new style to use?

```suggestion
    const source = this.source?.type;
```

See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining